didcomm/interfaces/did_doc.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from didcomm.types.types import JWK, KID, DID
+
+
+class VerificationMethod(ABC):
+    """Verification method."""
+
+    @abstractmethod
+    def as_jwk(self) -> JWK:
+        """Returns the JWK of this verification method."""
+        pass
+
+
+class DIDDoc(ABC):
+    """DID Document."""
+
+    @abstractmethod
+    def key_agreement(self, kid: KID) -> VerificationMethod:
+        """Gets keyAgreement verification method by the specified key ID."""
```

Comment:
What if there is no `key_agreement` for the specified KID? Should we raise an error? (the same for authentication and routing_keys)